File: Samples-Xamarin/Fragments/Chart/Series/ScatterBubbleSeriesFragment.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;
using Com.Telerik.Widget.Chart.Visualization.CartesianChart;
using Com.Telerik.Widget.Chart.Visualization.CartesianChart.Axes;
using Java.Util;
using Com.Telerik.Widget.Chart.Visualization.CartesianChart.Series.Scatter;
using Com.Telerik.Widget.Chart.Engine.Databinding;

namespace Samples
{
	public class ScatterBubbleSeriesFragment : Android.Support.V4.App.Fragment, ExampleFragment
	{
		public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
			ViewGroup rootView = (ViewGroup)inflater.Inflate(Resource.Layout.fragment_chart_example, container, false);
			rootView.AddView(this.createChart());
			return rootView;
		}

		private RadCartesianChartView createChart(){
			//Create the Chart View
			RadCartesianChartView chart = new RadCartesianChartView(this.Activity);

			//Create the scatter bubble series and attach axes and value bindings.
			ScatterBubbleSeries scatterBubbleSeries = new ScatterBubbleSeries();
			scatterBubbleSeries.BubbleScale = 1000;


			scatterBubbleSeries.XValueBinding = new XValueBinding();
			scatterBubbleSeries.YValueBinding = new YValueBinding();
			scatterBubbleSeries.BubbleSizeBinding = new AreaValueBinding();

			LinearAxis verticalAxis = new LinearAxis();
			//The values in the linear axis will not have values after the decimal point.
			verticalAxis.LabelFormat = "%.2f";
			LinearAxis horizontalAxis = new LinearAxis();
			horizontalAxis.LabelFormat = "%.2f";
			scatterBubbleSeries.VerticalAxis = verticalAxis;
			scatterBubbleSeries.HorizontalAxis = horizontalAxis;

			//Bind series to data
			scatterBubbleSeries.Data = this.getData();

			//Add series to chart
			chart.Series.Add(scatterBubbleSeries);
			return chart;
		}

		private ArrayList getData(){
			Java.Util.Random numberGenerator = new Java.Util.Random();
			ArrayList result = new ArrayList(8);

			for (int i = 0; i < 20; i++){
				ScatterDataEntity entity = new ScatterDataEntity();
				entity.value1 = numberGenerator.NextDouble() * 10;
				entity.value2 = numberGenerator.NextDouble() * 10;
				entity.bubbleArea = numberGenerator.NextInt(5) + 1;
				result.Add(entity);
			}

			return result;
		}

		public String Title() {
			return "Scatter bubble series";
		}

		public class XValueBinding : DataPointBinding{
			public override Java.Lang.Object GetValue (Java.Lang.Object p0)
			{
				ScatterDataEntity entity = (ScatterDataEntity)p0;
				return entity.value1;
			}

		}

		public class YValueBinding : DataPointBinding{
			public override Java.Lang.Object GetValue (Java.Lang.Object p0)
			{
				ScatterDataEntity entity = (ScatterDataEntity)p0;
				return entity.value2;
			}

		}

		public class AreaValueBinding : DataPointBinding{
			public override Java.Lang.Object GetValue (Java.Lang.Object p0)
			{
				ScatterDataEntity entity = (ScatterDataEntity)p0;
				return entity.bubbleArea;
			}

		}

		public class ScatterDataEntity : Java.Lang.Object{
			public double value1;
			public double value2;
			public int bubbleArea;
		}
	}
}

File: SDKExamples/Chart/src/main/java/fragments/series/ScatterBubbleSeriesFragment.java
package fragments.series;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import com.telerik.android.sdk.R;
import com.telerik.widget.chart.engine.databinding.FieldNameDataPointBinding;
import com.telerik.widget.chart.visualization.cartesianChart.RadCartesianChartView;
import com.telerik.widget.chart.visualization.cartesianChart.axes.LinearAxis;
import com.telerik.widget.chart.visualization.cartesianChart.series.scatter.ScatterBubbleSeries;
import com.telerik.widget.chart.visualization.cartesianChart.series.scatter.ScatterPointSeries;

import java.util.ArrayList;
import java.util.Random;

import activities.ExampleFragment;

/**
 * Created by ginev on 11/18/2014.
 */
public class ScatterBubbleSeriesFragment extends Fragment implements ExampleFragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        FrameLayout rootView = (FrameLayout)inflater.inflate(R.layout.fragment_scatter_bubble_series, container, false);
        rootView.addView(this.createChart());
        return rootView;
    }

    private RadCartesianChartView createChart(){
        //Create the Chart View
        RadCartesianChartView chart = new RadCartesianChartView(this.getActivity());

        //Create the bar series and attach axes and value bindings.
        ScatterBubbleSeries scatterBubbleSeries = new ScatterBubbleSeries();
        scatterBubbleSeries.setBubbleScale(1000);


        scatterBubbleSeries.setXValueBinding(new FieldNameDataPointBinding("value1"));
        scatterBubbleSeries.setYValueBinding(new FieldNameDataPointBinding("value2"));
        scatterBubbleSeries.setBubbleSizeBinding(new FieldNameDataPointBinding("bubbleArea"));

        LinearAxis verticalAxis = new LinearAxis();
        //The values in the linear axis will not have values after the decimal point.
        verticalAxis.setLabelFormat("%.2f");
        LinearAxis horizontalAxis = new LinearAxis();
        horizontalAxis.setLabelFormat("%.2f");
        scatterBubbleSeries.setVerticalAxis(verticalAxis);
        scatterBubbleSeries.setHorizontalAxis(horizontalAxis);

        //Bind series to data
        scatterBubbleSeries.setData(this.getData());

        //Add series to chart
        chart.getSeries().add(scatterBubbleSeries);
        return chart;
    }

    private ArrayList<ScatterDataEntity> getData(){
        Random numberGenerator = new Random();
        ArrayList<ScatterDataEntity> result = new ArrayList<ScatterDataEntity>(8);

        for (int i = 0; i < 20; i++){
            ScatterDataEntity entity = new ScatterDataEntity();
            entity.value1 = numberGenerator.nextDouble() * 10;
            entity.value2 = numberGenerator.nextDouble() * 10;
            entity.bubbleArea = numberGenerator.nextInt(5) + 1;
            result.add(entity);
        }

        return result;
    }

    @Override
    public String title() {
        return "Scatter bubble series";
    }

    public class ScatterDataEntity{
        public double value1;
        public double value2;
        public int bubbleArea;
    }
}
